src/mainloop.rkt

```diff
@@ -300,23 +300,13 @@
                     (list (*context*))))
   (cons domain (apply mk-pcontext pts+exs)))
 
-(define (initialize-alt-table! prog pcontext ctx)
-  (define alt (make-alt prog))
-  (*start-prog* prog)
-  (define table (make-alt-table (*pcontext*) alt ctx))
-
-  ; Add starting alt in every precision
-  (^table^
-   (if (*pareto-mode*)
-       (let ([new-alts (starting-alts alt ctx)])
-         (define-values (errss costs) (atab-eval-altns table new-alts ctx))
-         (atab-add-altns table new-alts errss costs))
-       table))
-
-  (when (flag-set? 'setup 'simplify)
-      (reconstruct! (patch-table-run-simplify (atab-active-alts (^table^))))
-      (finalize-iter!)
-      (^next-alts^ #f)))
+(define (initialize-alt-table! alternatives context pcontext)
+  (match-define (cons initial simplified) alternatives)
+  (*start-prog* (alt-expr initial))
+  (define table (make-alt-table pcontext initial context))
+  ;; TODO: Need to include starting-alts stuff here
+  (define-values (errss costs) (atab-eval-altns table simplified context))
+  (^table^ (atab-add-altns table simplified errss costs)))
```

Comment:
I think we can now add the `starting-alts` stuff back pretty easily. Basically `(startings-alts <alt>)` returns a list of alts, including the original and some variations (I think in different precisions?). So instead of adding "simplified" here you just want to add `(append-map starting-alts alternatives)`. Note that adding an alt that's already in the table, or adding duplicates, is basically free, so just make sure to pass this instead of `simplified` in both of these lines and you're good. (Or put it in a variable, whatever.)